test/component2_suggestors/componentwillmount_migrator_test.dart

```diff
@@ -46,255 +47,451 @@ componentWillMountTests({bool allowPartialUpgrades}) {
+            input: '''
+              @Component2()
+              class FooComponent extends UiComponent2 {
+                @override
+                componentWillMount() {
+                  var a = 1;
+                  var b = 2;
+                }
+                
+                @override
+                componentDidMount() {
+                  var c = 3;
+                  var d = 4;
+                }
+                
+                @override
+                componentWillUpdate() {}
+              }
+            ''',
+            expectedOutput: allowPartialUpgrades
+                ? '''
+              @Component2()
+              class FooComponent extends UiComponent2 {
+                @override
+                componentDidMount() {
+                  var c = 3;
+                  var d = 4;
+                  var a = 1;
+                  var b = 2;
+                }
+                
+                @override
+                componentWillUpdate() {}
+              }
+            '''
+                : '''
+              @Component2()
+              class FooComponent extends UiComponent2 {
+                @override
+                componentWillMount() {
+                  var a = 1;
+                  var b = 2;
+                }
+                
+                @override
+                componentDidMount() {
+                  var c = 3;
+                  var d = 4;
+                }
+                
+                @override
+                componentWillUpdate() {}
+              }
+            ''',
+          );
+        });
+      });
+
+      test('update super call to componentWillMount if not already existing',
+          () {
         testSuggestor(
-          expectedPatchCount: allowPartialUpgrades ? 1 : 0,
+          expectedPatchCount: 2,
           input: '''
             @Component2()
-            class FooComponent extends UiComponent2 {
-              componentWillMount(){
-                // method body
+            class FooComponent extends FluxUiComponent2 {
+              @override
+              void componentDidMount() {
+                var c = 3;
+                var d = 4;
               }
               
               @override
-              componentWillUnmount() {}
+              void componentWillMount() {
+                super.componentWillMount();
+                var a = 1;
+                var b = 2;
+              }
             }
           ''',
           expectedOutput: '''
             @Component2()
-            class FooComponent extends UiComponent2 {
-              ${allowPartialUpgrades ? 'init' : 'componentWillMount'}(){
-                // method body
-              }
-              
+            class FooComponent extends FluxUiComponent2 {
               @override
-              componentWillUnmount() {}
+              void componentDidMount() {
+                var c = 3;
+                var d = 4;
+                super.componentDidMount();
```

Comment:
I think in this case we also want the fixme comment, since this super-call might not be necessary anymore